scripts/seed_users.js

```diff
@@ -0,0 +1,414 @@
+    }
+
+    user.type = cleanString(user.type);
+    if (!user.type) {
+        user.type = "student";
+    }
+
+    // ToDo: Validate language from an array of valid languages
+    user.language = cleanString(user.language);
+    if (!user.language) {
+        user.language = "en";
+    }
+
+    user.color = cleanString(user.color);
+    if (!isValidColor(user.color)) {
+        // ToDo: Randomize Color
+        user.color = '{ "stroke":"#ff0000", "fill":"#0000ff" }';
+    }
+
+    // ToDo: Get minimum password length from .ini file
+    if (!user.password || typeof user.password != "string" || user.password.length < 4) {
+        console.log("Row: " + (index + 1) + " -> User: " + user.name + " -- Password is invalid\nGenerating random password...");
+        user.password = generatePassword(8);
+    }
+
+    user.classroom = user.classroom || "";
+    user.classroom = user.classroom.trim();
+
+    return user;
+}
+
+function stringifyUser(user) {
+    return '{"name":"' + user.name + '", ' + 
+            '"color":' + user.color + ', ' +
+            '"role":"' + user.type + '", ' +
+            '"password":"' + user.password + '", ' + 
+            '"language":"' + user.language + '", ' +
+            '"options":{"sync":true, "stats":true}}';
+}
+
+// Initiate Master Admin
+var masterAdmin = stringifyUser(new User('Master_Admin_' + (+new Date()).toString(), 'admin', 'en', '{"stroke":"#ff0000", "fill":"#0000ff"}', 'password'));
+
+// Insert User
+function insertUser(i) {
+    return new Promise(function(resolve, reject) {
+        request({
+            headers: {
+                "content-type": "application/json",
+                "x-access-token": masterAdmin.token,
+                "x-key": masterAdmin.user._id
+            },
+            json: true,
+            method: 'post',
+            uri: common.getAPIUrl() + 'api/v1/users',
+            body: {
+                user: stringifyUser(Users[i])
+            }
+        }, function(error, response, body) {
+            if (response.statusCode == 200) {
+                Users[i].status = 1;
+                Users[i]._id = body._id;
+                resolve(body);
+            } else {
+                Users[i].comment = body.error;
+                reject(body);
+            }
+        });
+    });
+}
+
+function generateCSV() {
+    console.log("generating csv...");
+}
+
+// Initiate Classrooms
+var Classrooms = {}; // Contains {classname: [students_array]}
+
+function stringifyClassroom(name) {
+    var newStudents = Classrooms[name].students;
+    var oldStudents = [];
+    if (Classrooms[name].data && Classrooms[name].data.students && Classrooms[name].data.students.length > 0) {
+        oldStudents = Classrooms[name].data.students 
+    }
+    var union = [...new Set([...newStudents, ...oldStudents])];
+    var students = JSON.stringify(union);
+    console.log(union);
+    return '{"name":"' + name + '","color":{"stroke":"#FF0000","fill":"#0000FF"},"students":' + students + '}';
```

Comment:
Will be good to randomize classroom color too.

Reply:
Done:
https://github.com/llaske/sugarizer-server/blob/c020bd58e37b2b48d7071161efdf24d67fec0505/scripts/seed_users.js#L219-L222

Color is existing classroom is retained.